srcpkgs/oilshell/template

```diff
@@ -0,0 +1,38 @@
+# Template file for 'oilshell'
+pkgname=oilshell
+version=0.6.0
+revision=1
+archs="x86_64*"
```

Comment:
What is the reason for restricting it to x86_64? It seems to compile fine for probably every arch, only cross compiling seems to be horribly broken to me.

Reply:
Because I couldn't get it compile on i686 with `./xbps-src -a i686 pkg oilshell` and I hadn't known about nocross. I didn't check armv* and aarch64 myselff, but it failed to build on travis for anything except x86_64 and x86_64-musl.
How's the new commit?

Reply:
Pls squash both commits into one.

Also `xbps-src` supports multiple masterdirs, so you can have a `x86_64`, `x86_64-musl` and `i686` at once and don't have to try to cross compile. (Check the `-m` flag, e..g `./xbps-src -m masterdir-i686 i686` will give you a i686 masterdir that can be used, and `./xbps-src -m masterdir-i686 pkg oilshell` would build you the package (same applies for musl)).

As a general rule you should make sure that everything that gets build by the build server is flagged correctly, but since `arm*` is only getting cross compiled you can stay under the assumption that it should work until proven otherwise.

Reply:
I made a separate masterdir for i686 and compiling there worked. I guess it just can't cross-compile..? I removed nocross from the file.
The git history is a bit of a mess since I'm still fairly new to using git itself, but everything should be squished now.

Reply:
It's failing on armv* and aarch in the spot where it tries to strip the binaries, though it successfully compiles on i686 now (yay!). Since xbps clearly isn't stripping the binary (if it does x86_64 fails) should I do something like `archs="x86_64* i686*"`?

Reply:
As I said, ad `nocross`

Reply:
ohh I was misunderstanding what `nocross` actually did. I've re-added it and travis finished the builds and the armv*/aarch builds no longer fail.
I also figured out why making readline an option wasn't working and fixed that. Thanks for being patient with me!

Reply:
Shouldn't readline be the default options. So you imo should add it to the `default_build_options`

Reply:
Good point, changed `build_options` to `build_options_default`.

Reply:
You need to have both `build_options` and `build_options_default`

Reply:
That would certainly explain why running `./xbps-src show-options oilshell` no longer worked. Fixed!